src/js/libgeo/GeoOps.js

```diff
@@ -371,6 +441,31 @@ geoOps.Free.updatePosition = function(el) {
     putStateComplexVector(param); // copy param
     el.homog = General.withUsage(param, "Point");
 };
+geoOps.Free.getRandomMove = function(el) {
+    var rcomplex = function() {
+        var fact = 0.1;
+        return CSNumber.complex((Math.random() - 0.5) * fact, (Math.random() - 0.5) * fact);
+    };
+    var oldpos = List.normalizeMax(el.homog);
+    var oz = oldpos.value[2];
+    var ozabs = CSNumber.abs(oz).value.real;
+
+    var rvect = List.turnIntoCSList([rcomplex(), rcomplex(), CSNumber.real(0)]);
+
+    // far points 
+    if (ozabs < CSNumber.eps) {
+        oz = CSNumber.real(Math.random());
+        rvect.value[2] = CSNumber.real(Math.random());
```

Comment:
This is “slightly evil”, since it modifies a CindyJS data structure and those are considered immutable after construction, in order to make copy-on-write work reliably. In this particular case things are safe, since we know that noone else has a copy of this `rvec.value` array. But nevertheless it would be cleaner to change the order of things: initialize some variable to zero, then change it to some random number in this `if` here, and only then construct the `rvec` taking that variable as the last coordinate.

What's the point in randomizing `oz` as well, as opposed to setting it to 1? If there is a good reason, a comment explaining it might be in order.